### PR TITLE
Add Python 3.11

### LINKED ISSUE #500: Add support for Python 3.11
We should add Python 3.11 to the test pipeline and fix any issues that are related to deprecations.

### PR DESCRIPTION
**Pull request by jkisse**
closes #500 

`np.bool` is deprecated and had to be replaced by bulit-in `bool`

numba does not support 3.11 yet

### CODE REVIEW

**Review by dlohmeier — APPROVED** [review]

File: .github/workflows/run_tests_develop.yml [hunk 1]
```diff
@@ -18,17 +18,18 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        python-version: ['3.7', '3.8', '3.9', '3.10']
```
Comment (SimonRubenDrauz) [inline .github/workflows/run_tests_develop.yml:21]: could you also add this to release.yml

File: .github/workflows/run_tests_master.yml [hunk 2]
```diff
@@ -17,18 +17,19 @@ jobs:
...
-        uses: actions/setup-python@v2
+        uses: actions/setup-python@v4
         with:
           python-version: ${{ matrix.python-version }}
       - name: Install dependencies
         run: |
           python -m pip install --upgrade pip
-          python -m pip install pytest python-igraph pytest-split numba
+          python -m pip install pytest python-igraph pytest-split
+          if ${{ matrix.python-version != '3.11' }}; then python -m pip install numba; fi
```
[the remark this answers is not in the document]
Comment (jkisse) [reply .github/workflows/run_tests_master.yml:32]: no

File: .github/workflows/run_tests_develop.yml (hunk 1 above)
Comment (jkisse) [reply .github/workflows/run_tests_develop.yml:21]: I'll add it in another PR

File: .github/workflows/run_tests_develop.yml [hunk 3]
```diff
@@ -113,9 +115,9 @@ jobs:
       matrix:
         python-version: [ '3.8' ]
```
[the remark this answers is not in the document]
Comment (jkisse) [reply .github/workflows/run_tests_develop.yml:116]: I don't know, you wrote that part recently ;-)

File: .github/workflows/run_tests_master.yml [hunk 4]
```diff
@@ -81,9 +83,9 @@ jobs:
       matrix:
         python-version: [ '3.8' ]
```
[the remark this answers is not in the document]
Comment (jkisse) [reply .github/workflows/run_tests_master.yml:84]: I don't know, you wrote that part recently ;-)

[the remark this answers is not in the document]
Comment (SimonRubenDrauz) [reply .github/workflows/run_tests_master.yml:84]: I based it on linting. Can we update it also to 3.11?

**Pull request merged**